torchvision/prototype/transforms/_geometry.py

```diff
@@ -0,0 +1,130 @@
+from typing import Any, Dict, Tuple, Union
+
+import torch
+from torch.nn.functional import interpolate
+from torchvision.prototype.datasets.utils import SampleQuery
+from torchvision.prototype.features import BoundingBox, Image
+from torchvision.prototype.transforms import Transform
+
+
+class HorizontalFlip(Transform):
+    @staticmethod
+    def image(input: Image) -> Image:
+        return Image(input.flip((-1,)), like=input)
+
+    @staticmethod
+    def bounding_box(input: BoundingBox) -> BoundingBox:
+        x, y, w, h = input.convert("xywh").to_parts()
+        x = input.image_size[1] - (x + w)
+        return BoundingBox.from_parts(x, y, w, h, like=input, format="xywh")
+
+
+class Resize(Transform):
+    def __init__(
+        self,
+        size: Union[int, Tuple[int, int]],
+        *,
+        interpolation_mode: str = "nearest",
+    ) -> None:
+        super().__init__()
+        self.size = (size, size) if isinstance(size, int) else size
+        self.interpolation_mode = interpolation_mode
+
+    def get_params(self, sample: Any) -> Dict[str, Any]:
+        return dict(size=self.size, interpolation_mode=self.interpolation_mode)
+
+    @staticmethod
+    def image(input: Image, *, size: Tuple[int, int], interpolation_mode: str = "nearest") -> Image:
+        return Image(interpolate(input.unsqueeze(0), size, mode=interpolation_mode).squeeze(0), like=input)
```

Comment:
Again, we are assuming here that `image` is just an individual image, and we will have separate classes for video and batches of images?

Reply:
`Video` yes, batches of images no-ish. From our preliminary talk I got that we do not want to support batched images.

Reply:
Let's chat to make sure we are aligned there

Reply:
As discussed offline, I'll add a `Video` feature as soon as we land the first video dataset.